cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -146,14 +144,19 @@ public NamespaceMeta load(NamespaceId namespaceId) throws Exception {
    * @throws NamespaceAlreadyExistsException if the specified namespace already exists
    */
   @Override
-  @AuthEnforce(entities = "instanceId", enforceOn = InstanceId.class, actions = Action.ADMIN)
   public synchronized void create(final NamespaceMeta metadata) throws Exception {
     // TODO: CDAP-1427 - This should be transactional, but we don't support transactions on files yet
     Preconditions.checkArgument(metadata != null, "Namespace metadata should not be null.");
     NamespaceId namespace = metadata.getNamespaceId();
     if (exists(namespace)) {
       throw new NamespaceAlreadyExistsException(namespace);
     }
+    String ownerPrincipal = metadata.getConfig().getPrincipal();
+    if (ownerPrincipal != null) {
+      authorizationEnforcer.enforce(new KerberosPrincipalId(ownerPrincipal), authenticationContext.getPrincipal(),
+                                    Action.ADMIN);
+    }
+    authorizationEnforcer.enforce(namespace, authenticationContext.getPrincipal(), Action.ADMIN);
```

Comment:
We cannot since now the `AuthEnforce` can only enforce on the parameters. Only after the pr is merged, we can enforce on local variables.